NEWS.md

```diff
@@ -1,3 +1,12 @@
+## 2.4.0 2023-03-22
```

Comment:
Migrating to Java 17 is a breaking change so update module version to 3.0.0

and these was not released yet so replace date with in progress